.docker/Dockerfile-CPU-Base

```diff
@@ -3,38 +3,144 @@
 # ------------------------------------------------------------------
 # Ubuntu           18.04
 # OpenMPI          latest       (apt)
-# cmake            3.10         (git)
-# MKL              2018.4-057   (apt)
-# arrayfire        3.7.1        (git, CPU backend)
-# oneDNN           2.0          (git)
-# Gloo             b7e0906      (git)
-# libsndfile       4bdd741      (git)
+# cmake            3.10         (oficial binaries)
+# MKL              2020.4-912   (apt)
+# arrayfire        3.7.3        (git, CPU backend)
+# libsndfile       latest       (apt, v1.0.28-4)
+# oneDNN           master       (git)
+# Gloo             1da2117      (git)
 # FFTW             latest       (apt)
 # KenLM            4a27753      (git)
 # GLOG             latest       (apt)
 # gflags           latest       (apt)
-# python           3.6          (apt)
+# python3          latest       (apt)
+# boost            1.75.0       (source)
 # ==================================================================
 
-FROM ubuntu:18.04
+FROM ubuntu:18.04 as cpu_base_builder
 
-ENV APT_INSTALL="apt-get install -y --no-install-recommends"
-ENV MKLROOT="/opt/intel/mkl"
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apt-get update && \
-    DEBIAN_FRONTEND=noninteractive $APT_INSTALL \
+RUN apt-get update && apt-get install -y --no-install-recommends \
         build-essential \
         ca-certificates \
-        wget \
+        curl \
+        git \
+        g++
+        # for cmake
+        #zlib1g-dev libcurl4-openssl-dev \
+        # for MKL
+        #apt-transport-https \
+        #gpg-agent gnupg2 \
+        # ssh for OpenMPI
+        #openssh-server openssh-client \
+        ## gflags
+        #libgflags-dev libgflags2.2 \
+        ## for glog
+        #libgoogle-glog-dev libgoogle-glog0v5 \
+
+# Install boost 1.75
+RUN cd /tmp && \
+    curl -sSLo - https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.bz2 | tar --bzip2 -x && \
+    cd boost_1_75_0 && \
+    ./bootstrap.sh --prefix=/opt/boost && \
+    ./b2 && \
+    ./b2 install
+ENV LD_LIBRARY_PATH="/opt/boost/lib:$LD_LIBRARY_PATH"
+
+# cmake 3.10
+RUN curl -sSLo - https://cmake.org/files/v3.10/cmake-3.10.3-Linux-x86_64.tar.gz | tar -xz -C /opt/cmake --strip-components 1
+ENV PATH="/opt/cmake/bin:$PATH"
+# ==================================================================
+# arrayfire with CPU backend https://github.com/arrayfire/arrayfire/wiki/Build-Instructions-for-Linux
+# ------------------------------------------------------------------
+FROM cpu_base_builder as cpu_arrayfire
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Comment:
Shouldn't be needed if installing MKL (we do need fftw3-dev)

Reply:
Arrayfire is not needed?